lib/input_validators.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module InputValidators
+  def validate_selection(selection)
+    index = selection.to_i - 1
+
+    if invalid? selection
+      @console.output('Invalid character! Please select an integer from 1-9:')
+      select_space
+    elsif occupied? index
+      @console.output('Invalid move! Please select a free space:')
+      select_space
+    else
+      @board.record_move(@current_player.mark, index)
+    end
+  end
+
+  def invalid?(input)
+    /^([1-9])$/.match(input).nil?
+  end
+
+  def occupied?(index)
+    @board.spaces[index] == @current_player.mark || @board.spaces[index] == @next_player.mark
+  end
```

Comment:
Now that I think about this more, this might be better as a responsibility of the `Board` class. It seems reasonable that a board knows if its space is occupied or not.